make credential format specific parameters madnatory in authorization details (#481)
openid/OpenID4VCI: openid-4-verifiable-credential-issuance-1_0.md

The following additional claims are defined for authorization details of type `openid_credential` and this Credential Format.

* `claims`: OPTIONAL. An array of claims description objects as defined in (#claims-description-authorization-details).
* `credential_definition`: REQUIRED. Object containing a detailed description of the Credential consisting of the following parameter:

>   * `type`: REQUIRED. Array as defined in (#server-metadata-jwt-vc-json). This claim contains the type values the Wallet requests authorization for at the Credential Issuer. It MUST be present if the claim `format` is present in the root of the authorization details object. It MUST not be present otherwise.

Reviewer: ```suggestion
  * `type`: REQUIRED if the `format` claim is present with a value of `jwt_vc_json`. It MUST NOT be present otherwise. Array as defined in (#server-metadata-jwt-vc-json). This claim contains the type values the Wallet requests authorization for at the Credential Issuer.
```

The following additional claims are defined for authorization details of type `openid_credential` and this Credential Format.  

* `claims`: OPTIONAL. An array of claims description objects as defined in (#claims-description-authorization-details).
* `credential_definition`: REQUIRED. Object containing the detailed description of the Credential. It consists of the following parameters:

>     * `@context`: REQUIRED. Array as defined in (#server-metadata-ldp-vc). It MUST only be present if the `format` claim is present in the root of the authorization details object. It MUST not be present otherwise. 

Reviewer: ```suggestion
    * `@context`: REQUIRED if the `format` claim is present with a value of `ldp_vc`. It MUST NOT be present otherwise. Array as defined in (#server-metadata-ldp-vc). 
```

>     * `type`: REQUIRED. Array as defined in (#server-metadata-ldp-vc).  This claim contains the type values the Wallet requests authorization for at the Credential Issuer. MUST only be present if the `@context` claim is present. 

Reviewer: ```suggestion
    * `type`: REQUIRED if the `format` claim is present with a value of `ldp_vc`. It MUST NOT be present otherwise. Array as defined in (#server-metadata-ldp-vc).  This claim contains the type values the Wallet requests authorization for at the Credential Issuer.
```

The following additional claims are defined for authorization details of type `openid_credential` and this Credential Format.

> * `doctype`: REQUIRED. String as defined in (#server-metadata-mso-mdoc). This claim contains the type value the Wallet requests authorization for at the Credential Issuer. It MUST only be present if the `format` claim is present. It MUST not be present otherwise. 

Reviewer 2: It can't be both "REQUIRED" and "MUST only be present if the `format` claim is present"?

Reviewer: Agreed, the other option we could do is have something like CONDITIONAL for these sorts of elements?

Author: we do not use conditional in openid4vc specs, because it's not part of rfc 2119

Author: ```suggestion
* `doctype`: REQUIRED when the `format` claim is present. MUST NOT be present otherwise. String as defined in (#server-metadata-mso-mdoc) that contains the type value the Wallet requests authorization for at the Credential Issuer.
```

Reviewer: ```suggestion
* `doctype`: REQUIRED if the `format` claim is present with a value of `mso_mdoc`. It MUST NOT be present otherwise. String as defined in (#server-metadata-mso-mdoc) that contains the type value the Wallet requests authorization for at the Credential Issuer.
```

Shouldn't we be more specific, also the example authorisation details object needs the format claim.

Reviewer: I've applied my suggestion above consistently, only other change is to adjust the examples which I'm happy to do.

Author: thank you!
